panic: runtime error: invalid memory address or nil pointer dereference - when a schema file contains text: null

```bash
kubeconform -v
v0.7.0
```

```bash
kubeconform -schema-location default -schema-location '.kubeconform/schema/{{.Group}}/{{.ResourceKind}}_{{.ResourceAPIVersion}}.json' repro.yaml
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x5ff9d7]

goroutine 10 [running]:
github.com/santhosh-tekuri/jsonschema/v6.(*roots).validate(0xc00036a240, 0xc0002646f0, {0x0, 0x0}, {0x0, 0x0})
        github.com/santhosh-tekuri/jsonschema/v6@v6.0.1/roots.go:241 +0xb7
github.com/santhosh-tekuri/jsonschema/v6.(*roots).addRoot(0xc00036a240, {0xc00048a140, 0x4c}, {0x0, 0x0})
        github.com/santhosh-tekuri/jsonschema/v6@v6.0.1/roots.go:53 +0x1e5
github.com/santhosh-tekuri/jsonschema/v6.(*roots).orLoad(0xc00036a240, {0xc00048a140, 0x4c})
        github.com/santhosh-tekuri/jsonschema/v6@v6.0.1/roots.go:38 +0x85
github.com/santhosh-tekuri/jsonschema/v6.(*roots).resolveFragment(0xc00026ca00?, {{0xc00048a140?, 0xc00048a0a0?}, {0x0?, 0xc00036a240?}})
        github.com/santhosh-tekuri/jsonschema/v6@v6.0.1/roots.go:63 +0x27
github.com/santhosh-tekuri/jsonschema/v6.(*Compiler).Compile(0xc000264600, {0xc00026ca00?, 0x8feaa8?})
        github.com/santhosh-tekuri/jsonschema/v6@v6.0.1/compiler.go:183 +0x4a
github.com/yannh/kubeconform/pkg/validator.downloadSchema({0xc00030f6c0?, 0x3?, 0x3?}, 0xc000376f00, {0xc00011a400, 0x9}, {0xc000120720, 0x1c}, {0x8fddde, 0x6})
        github.com/yannh/kubeconform/pkg/validator/validator.go:382 +0x3fa
github.com/yannh/kubeconform/pkg/validator.(*v).ValidateResource(0xc000371710, {{0x7ffe6bd684a9, 0xa}, {0xc00037a1e0, 0xeb, 0xf0}, 0xc00036a1c0, {0x0, 0x0}})
        github.com/yannh/kubeconform/pkg/validator/validator.go:209 +0x5ef
main.kubeconform.func1(0xc0000c6a10, 0xc0000c6930, {0x9b30e0, 0xc000371710})
        github.com/yannh/kubeconform/cmd/kubeconform/main.go:114 +0x12c
created by main.kubeconform in goroutine 1
        github.com/yannh/kubeconform/cmd/kubeconform/main.go:112 +0x5f2
```

repro.yaml
```yaml
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: route
spec:
  hostnames: []
  parentRefs:
  - name: gateway
    sectionName: https
  rules:
  - backendRefs:
    - kind: Service
      name: svc
      port: 80
```

```bash
cat .kubeconform/schema/gateway.networking.k8s.io/httproute_v1.json
null
```

--- a/pkg/validator/validator.go
+++ b/pkg/validator/validator.go
@@ -372,6 +372,13 @@ func downloadSchema(registries []registry.Registry, l jsonschema.SchemeURLLoader
 	for _, reg := range registries {
 		path, s, err = reg.DownloadSchema(kind, version, k8sVersion)
 		if err == nil {
+			// A schema document that decodes to nil (for example a file whose
+			// entire content is the literal "null") is accepted by AddResource
+			// but makes Compile dereference a nil root and panic. Treat it as a
+			// non-parseable response and try the next registry instead.
+			if s == nil {
+				continue
+			}
 			c := jsonschema.NewCompiler()
 			c.RegisterFormat(&jsonschema.Format{"duration", validateDuration})
 			c.UseLoader(l)

--- a/pkg/validator/validator_test.go
+++ b/pkg/validator/validator_test.go
@@ -460,6 +460,40 @@ interval: test
 			Invalid,
 			[]ValidationError{{Path: "/interval", Msg: "'test' is not valid duration: must start with P"}},
 		},
+		{
+			// A schema file whose entire content is the literal "null" decodes to
+			// a nil document. It used to reach jsonschema's Compile and panic with
+			// a nil pointer dereference (issue #337); it must now be treated as a
+			// missing schema instead.
+			"schema document is null",
+			[]byte(`
+kind: name
+apiVersion: v1
+firstName: foo
+`),
+			[]byte(`null`),
+			nil,
+			true,
+			false,
+			Skipped,
+			[]ValidationError{},
+		},
+		{
+			// Same as above but with IgnoreMissingSchemas disabled: a null schema
+			// document must surface as a graceful Error, not a panic (issue #337).
+			"schema document is null, do not ignore missing",
+			[]byte(`
+kind: name
+apiVersion: v1
+firstName: foo
+`),
+			[]byte(`null`),
+			nil,
+			false,
+			false,
+			Error,
+			[]ValidationError{},
+		},
 	} {
 		val := v{
 			opts: Opts{

